openapi.yaml

```diff
@@ -4711,16 +4711,32 @@ components:
       example: null
       description: |-
         The namespace the `process_id` is valid for.
+
+        The following options are pre-defined by the openEO API, but additional
+        namespaces may be introduced by back-ends or in a future version of the API.
         
-        By default, the namespace is `null`, which includes all pre-defined processes listed at `GET /processes`.
-        The namespace `user` refers to all user-defined processes listed as `GET /process_graphs`.
-        Additional namespaces may be introduced by back-ends or in a future version of the API.
+        * `null` (default): Checks both user-defined and pre-defined processes,
+           but prefers user-defined processes if both are available.
+           This allows users to add missing pre-defined processes for portability,
+           e.g. common processes from [processes.openeo.org](https://processes.openeo.org)
+           that have a process graph included.
+           If a user-defined process is prioritized over a pre-defined process,
+           a log entry SHOULD be added.
```

Comment:
about adding log entries: wouldn't it be better to *recommend* to always log the resolution (so not only when UDP is picked). Because if user finds no log: it could mean two things:
- UDP was picked but backend did not implement the logging (it's a SHOULD)
- PDP was picked

Reply:
recommend vs should: According to the RFC, it has the same meaning.

I have no strong reason against adding/changing that. How would you express that?

Reply:
ok I didn't realize that, to me "recommend" sound bit less forcing than "should".

Reply:
Also the sentence should state that it's  back-end-side logging

Reply:
suggestion:

> The back-end SHOULD add a log entry about whether the user-defined process or the predefined-process was picked to simplify post-mortem analysis.

minor note: I would tend to use MAY instead of SHOULD however, because the logging is not a first class feature of the API at this point, I would say

Reply:
> To me, "recommend" sound bit less forcing than "should".

Yeah, same for me.

> Also the sentence should state that it's back-end-side logging

Hmm... "a log entry SHOULD be added to corresponding back-end resource"?

Reply:
> The backend SHOULD add a log entry to the corresponding resource about whether the user-defined process or pre-defined process was picked, for debugging reasons.

Reply:
And for general logging, maybe: 

> It is RECOMMENDED to log the selected namespace for each process to the corresponding back-end resource.

Reply:
Oh, haven't seen yours early enough. Which one do you prefer? Mine seems to be a bit shorter...